hydroflow_lang/src/graph/ops/partition.rs

```diff
@@ -0,0 +1,408 @@
+                    )
+            });
+            let arg2_val = quote_spanned! {arg2_span.span()=> [ #( #idx_ints ),* ] };
+
+            (closure_idents, arg2_val)
+        } else {
+            // All indices.
+            let numeric_idents = (0..output_ports.len())
+                .map(|i| wc.make_ident(format!("{}_push", i)))
+                .collect();
+            let len_lit = LitInt::new(&format!("{}_usize", output_ports.len()), op_span);
+            let arg2_val = quote_spanned! {op_span=> #len_lit };
+            (numeric_idents, arg2_val)
+        };
+
+        let err_str = LitStr::new(
+            &format!(
+                "Index `{{}}` returned by `{}(..)` closure is out-of-bounds.",
+                op_name
+            ),
+            op_span,
+        );
+        let ident_item = wc.make_ident("item");
+        let ident_index = wc.make_ident("index");
+        let ident_unknown = wc.make_ident("match_unknown");
+
+        let sorted_outputs = output_sort_permutation.into_iter().map(|i| &outputs[i]);
+
+        let write_iterator = quote_spanned! {op_span=>
+            let #ident = {
+                #root::pusherator::demux::Demux::new(
+                    |#ident_item, #root::var_args!( #( #output_idents ),* )| {
+                        #[allow(unused_imports)]
+                        use #root::pusherator::Pusherator;
+
+                        let #ident_index = {
+                            #[allow(clippy::redundant_closure_call)]
+                            (#func)(&#ident_item, #arg2_val)
+                        };
+                        match #ident_index {
+                            #(
+                                #idx_ints => #output_idents.give(#ident_item),
+                            )*
+                            #ident_unknown => panic!(#err_str, #ident_unknown),
+                        };
+                    },
+                    #root::var_expr!( #( #sorted_outputs ),* ),
+                )
+            };
+        };
+
+        Ok(OperatorWriteOutput {
+            write_iterator,
+            ..Default::default()
+        })
+    },
+};
+
+/// Returns `Ok(Some(idents))` if ports are idents, or `Ok(None)` if ports are indices.
+/// Returns `Err(())` if there are any errors (pushed to `diagnostics`).
+fn determine_indices_or_idents(
+    output_ports: &[PortIndexValue],
+    op_span: Span,
+    op_name: &'static str,
+    diagnostics: &mut Vec<Diagnostic>,
+) -> Result<Option<Vec<Ident>>, ()> {
+    // Port idents supplied via port connections in the surface syntax.
+    // Two modes, either all numeric `0, 1, 2, 3, ...` or all `Ident`s.
+    // If ports are `Idents` then the closure's 2nd argument, input array must have named
+    // values corresponding to the port idents.
+    let mut ports_numeric = BTreeSet::new();
+    let mut ports_idents = Vec::new();
+    // If any ports are elided we return `Err(())` early.
```

Comment:
"elided" is a fancy word and kind of vague. is an "elided" port an expression like the second line below?
```
demux_out[a] -> ...;
demux_out -> ... ;
```
Maybe say "if a demux operator is used without naming an output port" or something?